Tests/SwiftParserTest/AttributeTests.swift

```diff
@@ -251,6 +251,61 @@ final class AttributeTests: XCTestCase {
     )
   }
 
+  func testIgnoreDeprecationAttribute() throws {
+    AssertParse("""
+    @available(*, deprecated)
+    class XClass {
+      var num: Int
+      init(_ num: Int) { self.num = num }
+    }
+    
+    @ignoreDeprecations(XClass)
+    func x() {
+      print(XClass(3))
+    }
+    """)
+
+    AssertParse("""
+    struct DeviceConf: Codable {
+
+      enum DeviceStatus: Codable {
+        case sandboxed
+        case partiallyEscaped
+        case fullyUnsandboxed
+      }
+
+      var isAMFIEnabled: Bool
+      var bootArgs: [String]
+
+      @available(*, deprecated)
+      var status: DeviceStatus
+
+      @ignoreDeprecation(status)
+      var canStartSession: Bool {
+        return status == .fullyUnsandboed
+      }
+    }
+    
+    """)
+
+    AssertParse("""
+    
+    @available(*, deprecated)
+    func deprecatedFunction() -> Int32 {
+      return 3
+    }
+    
+    @ignoreDeprecation(deprecatedFunction)
+    class StateManager {
+    
+      func someFunction() {
+        print(deprecatedFunction())
+      }
+    }
+    """)
```

Comment:
Would be nice to also have a test for `import`, `subscript`, `extension`, and enum elements. Note that they don't need to be fully complete Swift programs as we're only checking that they parse correctly (there's no harm in there being extra code tho).